packages/template-retail-react-app/app/ssr.js

```diff
@@ -33,19 +34,64 @@ const options = {
 const runtime = getRuntime()
 
 const {handler} = runtime.createHandler(options, (app) => {
+    const getRuntimeEnv = () => {
+        if (process.env.NODE_ENV !== 'production') return process.env.NODE_ENV ?? 'development'
+        const origin = getAppOrigin()
+        // mobify-storefront-staging sites have NODE_ENV set to production, but for the purposes
+        // of CSP we consider the sites to be staging.
+        return origin.endsWith('.mobify-storefront-staging.com') ? 'staging' : 'production'
+    }
+
+    // This is a temporary solution while we work on Storefront Preview - a full solution will
+    // prevent required CSP headers from being modified.
```

Comment:
This is probably not the final CSP we want to release. It's a decent sized block of new code in a file that's widely customized, but most of the code changes are to facilitate our own, internal development process.

(Not a blocking comment, though.)